test/common/protobuf/utility_test.cc

```diff
@@ -147,15 +147,31 @@ TEST_F(ProtobufUtilityTest, LoadBinaryProtoFromFile) {
   EXPECT_TRUE(TestUtility::protoEqual(bootstrap, proto_from_file));
 }
 
+// An unknown field (or with wrong type) in a message is rejected.
 TEST_F(ProtobufUtilityTest, LoadBinaryProtoUnknownFieldFromFile) {
   ProtobufWkt::Duration source_duration;
   source_duration.set_seconds(42);
   const std::string filename =
       TestEnvironment::writeStringToFileForTest("proto.pb", source_duration.SerializeAsString());
   envoy::config::bootstrap::v2::Bootstrap proto_from_file;
-  EXPECT_THROW_WITH_MESSAGE(
-      TestUtility::loadFromFile(filename, proto_from_file, *api_), EnvoyException,
-      "Protobuf message (type envoy.config.bootstrap.v2.Bootstrap) has unknown fields");
+  EXPECT_THROW_WITH_MESSAGE(TestUtility::loadFromFile(filename, proto_from_file, *api_),
+                            EnvoyException,
+                            "Protobuf message (type envoy.config.bootstrap.v2.Bootstrap with "
```

Comment:
nit: this error message is a bit redundant with multiple "unknown fields." Any way to clean it up a bit with the new logic?

Reply:
I've changed this to `unknown field set {1, 2}`. It's a bit hard to avoid the double unknown, since the JSON throw has an opaque string with its own description that needs wrapping to explain that it relates to unknown fields.